python/lsst/ap/association/diaPipe.py

```diff
@@ -298,9 +299,10 @@ def __init__(self, initInputs=None, **kwargs):
 
     def runQuantum(self, butlerQC, inputRefs, outputRefs):
         inputs = butlerQC.get(inputRefs)
-        expId, expBits = butlerQC.quantum.dataId.pack("visit_detector",
-                                                      returnMaxBits=True)
-        inputs["ccdExposureIdBits"] = expBits
+        inputs["idGenerator"] = self.config.idGenerator.apply(butlerQC.quantum.dataId)
+        # Need to set ccdExposureIdBits (now deprecated) to None and pass it,
+        # since there are non-optional positional arguments after it.
+        inputs["ccdExposureIdBits"] = None
```

Comment:
Since we have to change the API here anyway, should we create the `idGenerator` inside of `run` instead, so nothing has to be passed in? I was always a bit skeptical of this modification of `inputs`.

Reply:
(see related replies in pipe_tasks and in the main PR thread)